src/library/scala/math/PartiallyOrdered.scala

```diff
@@ -18,33 +17,22 @@ package math
  */
 trait PartiallyOrdered[+A] {
 
+  type AsPartiallyOrdered[B] = B => PartiallyOrdered[B]
+
   /** Result of comparing `'''this'''` with operand `that`.
    *  Returns `None` if operands are not comparable.
    *  If operands are comparable, returns `Some(x)` where
    *  - `x < 0`    iff   `'''this''' &lt; that`
    *  - `x == 0`   iff   `'''this''' == that`
    *  - `x > 0`    iff   `'''this''' &gt; that`
    */
-  def tryCompareTo [B >: A <% PartiallyOrdered[B]](that: B): Option[Int]
-
-  def <  [B >: A <% PartiallyOrdered[B]](that: B): Boolean =
-    (this tryCompareTo that) match {
-      case Some(x) if x < 0 => true
-      case _ => false
-    }
-  def >  [B >: A <% PartiallyOrdered[B]](that: B): Boolean =
-    (this tryCompareTo that) match {
-      case Some(x) if x > 0 => true
-      case _ => false
-    }
-  def <= [B >: A <% PartiallyOrdered[B]](that: B): Boolean =
-    (this tryCompareTo that) match {
-      case Some(x) if x <= 0 => true
-      case _ => false
-    }
-  def >= [B >: A <% PartiallyOrdered[B]](that: B): Boolean =
-    (this tryCompareTo that) match {
-      case Some(x) if x >= 0 => true
-      case _ => false
-    }
+  def tryCompareTo[B >: A : AsPartiallyOrdered](that: B): Option[Int]
+
+  def <  [B >: A : AsPartiallyOrdered](that: B): Boolean = (this tryCompareTo that) exists (_ < 0)
+
+  def >  [B >: A : AsPartiallyOrdered](that: B): Boolean = (this tryCompareTo that) exists (_ > 0)
+
+  def <= [B >: A : AsPartiallyOrdered](that: B): Boolean = (this tryCompareTo that) exists (_ <= 0)
+
+  def >= [B >: A : AsPartiallyOrdered](that: B): Boolean = (this tryCompareTo that) exists (_ >= 0)
```

Comment:
Let's keep this PR focused on removing deprecation. It doesn't look like the inliner is able to recover the code we had before. Not sure why /cc @lrytz 

From `wget https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots/org/scala-lang/scala-library/2.13.0-pre-889443e-SNAPSHOT/scala-library-2.13.0-pre-889443e-20170830.112331-1.jar ; unzip scala-library-2.13.0-pre-889443e-20170830.112331-1.jar scala/math/PartiallyOrdered.class ; cfr scala/math/PartiallyOrdered.class`

```
    default public <B> boolean $greater$eq(B that, Function1<B, PartiallyOrdered<B>> evidence$5) {
        return this.tryCompareTo(that, evidence$5).exists(x$4 -> {
            if (x$4 >= 0) {
                return true;
            }
            return false;
        }
        );
    }
```

Reply:
That's a good question. If I compile https://github.com/scala/scala/commit/889443e locally using 2.13.0-M2 (the starr) and pass `-opt:l:inline '-opt-inline-from:**'`, then inlining works correctly.

Reply:
If I run `sbt 'setupPublishCore https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots' publishLocal`, then it's not inlined. I'll dig into this.

Reply:
Mistery solved: https://github.com/scala/scala-dev/issues/424

Reply:
:+1: